components/dashboard/src/data/git-providers/unified-repositories-search-query.ts

```diff
@@ -9,27 +9,52 @@ import { useSearchRepositories } from "./search-repositories-query";
 import { useSuggestedRepositories } from "./suggested-repositories-query";
 import { useMemo } from "react";
 
+type UnifiedRepositorySearchArgs = {
+    searchString: string;
+    // If true, excludes projects and only shows 1 entry per repo
+    excludeProjects?: boolean;
+};
 // Combines the suggested repositories and the search repositories query into one hook
-export const useUnifiedRepositorySearch = ({ searchString }: { searchString: string }) => {
+export const useUnifiedRepositorySearch = ({ searchString, excludeProjects = false }: UnifiedRepositorySearchArgs) => {
     const suggestedQuery = useSuggestedRepositories();
     const searchQuery = useSearchRepositories({ searchString });
 
     const filteredRepos = useMemo(() => {
-        const repoMap = new Map<string, SuggestedRepository>(
-            (suggestedQuery.data || []).map((r) => [`${r.url}:${r.projectId || ""}`, r]),
-        );
+        const repoMap = new Map<string, SuggestedRepository>();
+
+        // Add suggested results to map
+        for (const repo of suggestedQuery.data || []) {
```

Comment:
To reduces code duplication and makes the logic a bit clearer. Furthermore, it slightly enhances performance by not looping through the repositories twice and, instead, simply looping through a two-element array containing the original data arrays. This is more efficient because the data arrays might be significantly larger than the two-element array.

```suggestion
        const processRepo = (repo: SuggestedRepository, excludeProjects: boolean) => {
```

Reply:
Or instead of that to handle empty data

```suggestion
        const processRepos = (repos: SuggestedRepository[]) => {
            for (const repo of repos) {
```

Reply:
Resolved in 563c635d9139b10ac724a201eafc284d108b0a61